week1/homework/src/server.js

```diff
@@ -5,13 +5,38 @@ const http = require('http');
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
+
 function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
-  });
+    if (request.url.startsWith('/state')) {
+      response.writeHead(200, { 'Content-Type': 'application/json' });
+      response.write(JSON.stringify({ state: state }));
+    }
+ else if (request.url === '/add') {
+      state = state + 1;
+      response.writeHead(200, { 'Content-Type': 'application/json' });
+      response.write(JSON.stringify({ state: state }));
+    }
+ else if (request.url === '/subtract') {
+      state = state - 1;
+      response.writeHead(200, { 'Content-Type': 'application/json' });
+      response.write(JSON.stringify({ state: state }));
```

Comment:
Over here - you can make use of the ES6 shorthand syntax instead of writing as { state: state }. Please have a look here - https://alligator.io/js/object-property-shorthand-es6/

Reply:
Oh, I am not aware of that, thanks for it. It is ok now.